presto-raptorx/src/main/java/com/facebook/presto/raptorx/metadata/DatabaseMetadataWriter.java

```diff
@@ -0,0 +1,718 @@
+
+            List<ColumnInfo> columns = dao.getColumnInfo(tableId);
+            checkConflict(columns.stream().anyMatch(column -> column.getColumnId() == columnId), "Column no longer exists");
+            checkConflict(columns.size() > 1, "Table only has one column");
+
+            deleteColumn(dao, commitId, tableId, columnId);
+
+            updateTable(dao, commitId, tableId);
+        });
+    }
+
+    @Override
+    public void createView(long commitId, ViewInfo view)
+    {
+        writeMaster(commitId, (dao, commit) -> {
+            failIfRelationExists(dao, view.getSchemaId(), view.getViewName());
+            dao.insertView(
+                    commitId,
+                    view.getViewId(),
+                    view.getViewName().getBytes(UTF_8),
+                    view.getSchemaId(),
+                    view.getCreateTime(),
+                    view.getUpdateTime(),
+                    view.getViewData().getBytes(UTF_8),
+                    utf8Bytes(view.getComment()));
+        });
+    }
+
+    @Override
+    public void dropView(long commitId, long viewId)
+    {
+        writeMaster(commitId, (dao, commit) -> {
+            checkConflict(dao.viewIdExists(viewId), "View no longer exists");
+            deleteView(dao, commitId, viewId);
+        });
+    }
+
+    @Override
+    public void insertChunks(long commitId, long tableId, Collection<ChunkInfo> chunks)
+    {
+        Set<Long> columnIds = chunks.stream()
+                .map(ChunkInfo::getColumnStats)
+                .flatMap(Collection::stream)
+                .map(ColumnStats::getColumnId)
+                .collect(toImmutableSet());
+
+        long currentTime = System.currentTimeMillis();
+        AtomicReference<List<ColumnInfo>> columns = new AtomicReference<>();
+        AtomicReference<TableInfo> table = new AtomicReference<>();
+
+        writeMaster(commitId, (dao, commit) -> {
+            checkConflict(dao.tableIdExists(tableId), "Table no longer exists");
+
+            table.set(dao.getTableInfo(tableId).builder()
+                    .addRowCount(chunks.stream().mapToLong(ChunkInfo::getRowCount).sum())
+                    .setUpdateTime(currentTime)
+                    .build());
+
+            columns.set(dao.getColumnInfo(tableId));
+            Set<Long> tableColumnIds = columns.get().stream()
+                    .map(ColumnInfo::getColumnId)
+                    .collect(toSet());
+            checkConflict(tableColumnIds.containsAll(columnIds), "Column no longer exists");
+
+            deleteTable(dao, commitId, tableId);
+            insertTable(dao, commitId, table.get());
+
+            dao.updateRollbackInfo(commit.getRollbackInfo()
+                    .withWrittenTable(tableId)
+                    .toBytes());
+        });
+
+        writeShard(commitId, getTableShard(tableId), dao -> {
+            dao.insertChunks(commitId, tableId, currentTime, chunks.stream()
```

Comment:
Shall we consider split chunks into multiple mysql transactions? Otherwise this line can still exceed mysql transaction limit.

And it's safe to split, right?  Each chunk has the CommitID.

Reply:
Yes, I'll change this to use batches.